src/main/java/io/github/cottonmc/energy/api/EnergyExtractor.java

```diff
@@ -0,0 +1,40 @@
+package io.github.cottonmc.energy.api;
+
+import io.github.cottonmc.energy.util.ContainerInteraction;
+import io.github.cottonmc.energy.util.PacketTier;
+import net.minecraft.util.math.Direction;
+
+public interface EnergyExtractor {
```

Comment:
This probably wants to be called "EnergyProducer" for symmetry with "EnergyConsumer"